misc/semver-checker/modules/semver-checker-core/src/main/java/io/ballerina/semver/checker/central/CentralClientWrapper.java

```diff
@@ -0,0 +1,181 @@
+                        highestVersion));
+                return highestVersion;
+            } else {
+                LOGGER.debug("local changes will be compared against the release version: " + compatibleVersion.get());
+                return compatibleVersion.get();
+            }
+        } catch (CentralClientException e) {
+            throw new SemverToolException(e);
+        }
+    }
+
+    /**
+     * Pulls the specified package from the central.
+     *
+     * @param orgName organization name of the package
+     * @param pkgName name of the package
+     * @param version version of the package
+     * @return path of the pulled bala package
+     * @throws SemverToolException if unexpected error occurred while pulling package
+     */
+    public Path pullPackage(String orgName, String pkgName, SemanticVersion version) throws SemverToolException {
+        Path packagePathInBalaCache = ProjectUtils.createAndGetHomeReposPath()
+                .resolve(ProjectConstants.REPOSITORIES_DIR).resolve(ProjectConstants.CENTRAL_REPOSITORY_CACHE_NAME)
+                .resolve(ProjectConstants.BALA_DIR_NAME)
+                .resolve(orgName).resolve(pkgName);
+        try {
+            // avoids pulling from the central if the package version already available in the local repository.
+            if (Files.exists(packagePathInBalaCache) && Files.isDirectory(packagePathInBalaCache)) {
+                throw new PackageAlreadyExistsException("package already exists in the home repository: " +
+                        packagePathInBalaCache);
+            }
+
+            errStream.printf("pulling package version '%s/%s:%s' from central...%n", orgName, pkgName, version);
+            centralClient.pullPackage(orgName, pkgName, version.toString(), packagePathInBalaCache,
+                    SUPPORTED_PLATFORM_JAVA_11, RepoUtils.getBallerinaVersion(), false);
+        } catch (PackageAlreadyExistsException e) {
+            // can be ignored
+        } catch (CentralClientException e) {
+            throw new SemverToolException("unexpected error occurred while pulling package from the central: "
+                    + e.getMessage());
+        }
+
+        return packagePathInBalaCache.resolve(version.toString()).resolve(SUPPORTED_PLATFORM_JAVA_11);
+    }
+
+    /**
+     * Returns the "closest" compatible version for the given semantic version, from the provided list of versions.
+     *
+     * @param versionList list of available versions in the Ballerina central
+     * @param pkgVersion  target version
+     */
+    private Optional<SemanticVersion> selectCompatibleVersion(List<SemanticVersion> versionList,
+                                                              SemanticVersion pkgVersion) {
+        AtomicReference<SemanticVersion> latestCompatibleVersion = new AtomicReference<>(SemanticVersion.from("0.0.0"));
```

Comment:
Shal we use `ProjectConstants.DEFAULT_VERSION` instead of `"0.0.0"`?

Reply:
IMO we better stick with `0.0.0` because it's possible that the `ProjectConstants.DEFAULT_VERSION` can be changed to a different value in the future? (i.e. `0.0.1`)